### PR TITLE
Implement Clone Key action to AnimationPlayer

### PR DESCRIPTION
**Pull request by sofia285**
Added "Clone Key" button to the AnimationPlayer which allows you to duplicate and bind keys. Added "Make Unique" button, which breaks the connection between cloned keys.

Co-authored-by: Miguel Coelho <miguel.pinheiro.coelho@tecnico.ulisboa.pt>

### CODE REVIEW

File: doc/classes/Animation.xml [hunk 1]
```diff
@@ -424,6 +424,46 @@
...
 			</description>
 		</method>
+		<method name="add_key_to_bind">
+			<return type="int" />
+			<param index="0" name="bind_idx" type="int" />
+			<param index="1" name="p_at_pos" type="int" />
+			<param index="2" name="track_idx" type="int" />
+			<param index="3" name="key_idx" type="int" />
+			<description>
+				Binds a key to a list of binded keys. Returns the index of the added key in the list.
```
Comment (AThousandShips) [inline doc/classes/Animation.xml:434]:
```suggestion
				Binds a key to a list of bound keys. Returns the index of the added key in the list.
```
For all instances

File: scene/resources/animation.h [hunk 2]
```diff
@@ -115,7 +115,17 @@ class Animation : public Resource {
 		virtual ~Track() {}
 	};
 
+	struct BindedKey {
```
Comment (AThousandShips) [inline scene/resources/animation.h:118]:
```suggestion
	struct BoundKey {
```

File: scene/resources/animation.h [hunk 3]
```diff
@@ -236,6 +246,7 @@ class Animation : public Resource {
 	};
 
 	Vector<Track *> tracks;
+	Vector<BindParent *> binded_keys;
```
Comment (AThousandShips) [inline scene/resources/animation.h:249]:
```suggestion
	Vector<BindParent *> bound_keys;
```

File: scene/resources/animation.h [hunk 4]
```diff
@@ -115,7 +115,17 @@ class Animation : public Resource {
 		virtual ~Track() {}
 	};
 
+	struct BindedKey {
+		int track_idx;
+		int key_idx;
+	};
+
+	struct BindParent {
+		Vector<BindedKey *> bindedChildren;
```
[the remark this answers is not in the document]
Comment (sofia285) [reply scene/resources/animation.h]: Thanks for the advice, just changed it!

**Pull request closed without merging**